pr2eus/robot-interface.l

```diff
@@ -1186,6 +1187,26 @@ Return value is a list of interpolatingp for all controllers, so (null (some #'i
    )
   )
 ;;
+;; for text-to-speech services
+(defmethod robot-interface
+  (:speak (text &rest args &key (lang :en) &allow-other-keys)
+    "Speak sentence using text-to-speech services.
+     Args:
+        text: sentence to speak
+        lang: language to speak, currently :en or :ja are supported.
+        args: other key arguments are passed to speech functions"
+    (case lang
+     (:en (send* self :speak-en text (remprop :lang args)))
+     (:ja (send* self :speak-jp text (remprop :lang args)))
+     (t (error "Unsupported language ~A" lang))))
+  (:speak-en (text &rest args)
+    "Speak english sentence"
+    (apply #'speak-en text args))
+  (:speak-jp (text &rest args)
+    "Speak japanese sentence"
+    (apply #'speak-jp text args))
+) ;; defmethod robot-interface (text-to-speech)
```

Comment:
how about add `:(send-speak-msg` here and change
```
(defun speak-jp (text &rest args)
 (warning-message 3 "Function `speak-jp` will be deprecated, please use (send *ri* :speak \"text\" :lang :ja") 
 (send* *ri* :speak text :lang :ja args) ;; this is dangerous, because it assumes *ri*..... is this ok? if not leave current implementation here, just to show warning message"

Reply:
@k-okada Sorry for a late response.
That's indeed a good idea. I updated as followings:
- added warning message for deprecated functions `speak-jp` and `speak-en`.
- move text-to-speech into `robot-interface`
- call `robot-interface` method `:speak-jp` internally in `speak-jp` function. (same for `speak-en`)
- updated test codes